fcrepo-kernel-modeshape/src/main/java/org/fcrepo/kernel/modeshape/services/ContainerServiceImpl.java

```diff
@@ -61,10 +64,29 @@ public Container findOrCreate(final Session session, final String path) {
 
             if (node.isNew()) {
                 initializeNewObjectProperties(node);
+
+                final Node parent = node.getDepth() > 0 ? node.getParent() : null;
```

Comment:
If the `dsNode` was created without a `slug` and therefore the `parent` is a PairTree, do we want to be touching the PairTree or the logical parent container? I suspect the latter.